postcss-selector-parser.d.ts

```diff
@@ -311,6 +351,51 @@ declare namespace parser {
          * The attribute name after having been qualified with a namespace.
          */
         readonly qualifiedAttribute: string;
+
+        /**
+         * The case insensitivity flag or an empty string depending on whether this
+         * attribute is case insensitive.
+         */
+        readonly insensitiveFlag : 'i' | '';
+
+        /**
+         * Returns the attribute's value quoted such that it would be legal to use
+         * in the value of a css file. The original value's quotation setting
+         * used for stringification is left unchanged. See `setValue(value, options)`
+         * if you want to control the quote settings of a new value for the attribute or
+         * `set quoteMark(mark)` if you want to change the quote settings of the current
+         * value.
+         *
+         * You can also change the quotation used for the current value by setting quoteMark.
+         **/
+        quoteValue(options?: SmartQuoteMarkOptions): string;
+
+        /**
+         * Set the unescaped value with the specified quotation options. The value
+         * provided must not include any wrapping quote marks -- those quotes will
+         * be interpreted as part of the value and escaped accordingly.
+         * @param value
+         */
+        setValue(value: string, options?: SmartQuoteMarkOptions): void;
```

Comment:
Should `options` be `SmartQuoteMarkOptions | PreferredQuoteMarkOptions`? What if I want to explicitly set the quote type when setting the value?

Reply:
In fact, you could unify the two quote mark options types to a single type with all three as optional properties. That interface would make sense with the quote mark getters defined down below as well (if the two methods are unified as suggested).

Reply:
`SmartQuoteMarkOptions` extends `PreferredQuoteMarkOptions` and all values are optional. So the distinction between these two types is just nominal to make clear that for some methods, the `smart` option, if passed, is ignored.